docker/software/service/running.sls

```diff
@@ -38,6 +39,13 @@ docker-software-service-running-unmasked:
         {%- endif %}
 
 docker-software-service-running-docker:
+  {%- if 'environ' in d.pkg.docker and d.pkg.docker.environ %}
+  cmd.run:
+    - name: systemctl daemon-reload
+    - onchanges:
+      - sls: {{ sls_dropin }}
+  {%- endif %}
+
```

Comment:
I'm not sure it's a good idea to be able to have an empty state if `environ` is not set.

Shouldn't the if be move before declaring the state, the dependencies are well conditioned as I can the bellow?

Reply:
I believe you are referring to my `dropin.sls` state? I modeled it after the formula's existing `environ.sls`, which will also be a blank state if environ is not set.

Would you prefer I change those states to end with something like this?

```
  {%- else %}
docker-software-environ-file-managed-environ_file:
  test.succeed_without_changes:
    - name: No environ changes needed
  {%- endif %}

```

Reply:
No, I quoted the `docker-software-service-running-docker` state in `docker/software/service/running.sls`.

Reply:
Yes, but the `docker-software-service-running-docker` ID declaration is not blank - it will always have the `service.running` state declaration. I just might not have the `cmd.run` state declaration if `environ` is not set. This is my (perhaps paranoid) approach of _absolutely do not even consider running daemon-reload if it is not necessary_. It could potentially be done through non-jinja means with `onchanges`, but I wasn't sure the behavior of `onchanges` for a potentially empty sls like `- sls: {{ sls_dropin }}` worked reliably

Reply:
Ok, my bad, I only see the diff and not the full file

:+1: